stock_quant_package_dimension/models/stock_quant_package.py

```diff
@@ -20,6 +26,49 @@ class StockQuantPackage(models.Model):
         help="volume in cubic meters",
     )
 
+    def _get_picking_move_line_ids_per_package(self, picking_id):
+        if not picking_id:
+            return {}
+        move_lines = self.env["stock.move.line"].search(
+            [
+                ("result_package_id", "in", self.ids),
+                ("picking_id", "=", self.env.context["picking_id"]),
+            ]
+        )
+        res = dict.fromkeys(self.ids, self.env["stock.move.line"])
+        for ml in move_lines:
+            res.setdefault(ml.result_package_id, set(ml.ids))
+            res[ml.result_package_id].add(ml.id)
+        return res
+
+    def _get_weight_from_move_lines(self, move_lines):
+        return sum(
+            ml.product_uom_id._compute_quantity(ml.qty_done, ml.product_id.uom_id)
+            * ml.product_id.weight
+            for ml in move_lines
+        )
+
+    def _get_weight_from_quants(self, quants):
+        return sum(quant.quantity * quant.product_id.weight for quant in quants)
+
+    @api.depends("quant_ids")
+    @api.depends_context("picking_id")
+    def _compute_estimated_pack_weight(self):
+        # NOTE: copy-pasted and adapted from `delivery` module
+        # because we do not want to add the dependency against 'delivery' here.
+        move_line_ids_per_package = self._get_picking_move_line_ids_per_package(
+            self.env.context.get("picking_id")
+        )
+        for package in self:
+            weight = 0.0
+            if self.env.context.get("picking_id"):
+                move_line_ids = move_line_ids_per_package[package]
+                move_lines = self.env["stock.move.line"].browse(move_line_ids)
```

Comment:
```suggestion
        picking_id = self.env.context.get("picking_id")
        move_line_ids_per_package = self._get_picking_move_line_ids_per_package(
            picking_id
        )
        for package in self:
            weight = 0.0
            move_line_ids = move_line_ids_per_package.get(package)
            if move_line_ids:
                move_lines = self.env["stock.move.line"].browse(move_line_ids)
```

Reply:
I'm not sure about this one: with your implementation, if there is no move lines we will get the weight computed from quants, even if a picking_id was passed in the context.
But at the same time if we pass the picking_id there for sure move lines related (at least in Shopfloor, don't know the use case in `delivery` with the `weight` field).
So I'll replace your `if move_line_ids` with a `if picking_id`

Reply:
I thought it would be a good idea anyway to have the std weight when no move lines are found. Fine with your explanation :)